src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,164 @@
+package duke;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * Stores the list of tasks.
+ * Saves/loads tasks from a data file on shutdown/startup.
+ * Handles data manipulation
+ */
+public class Storage {
+
+    private static ArrayList<Task> taskArray = new ArrayList<>();
+
+    /**
+     * Loads tasks from data file if it exists
+     */
+    public Storage() {
```

Comment:
This constructor seems quite long with a lot of logic.

What do you think about abstracting the creation of the directory and writing of tasks into a private method in the same class?